### PR TITLE
Kasuba/partial demo

### PR DESCRIPTION
**Pull request by kasuba97**
rushed-merge
will cleaned up the double fetch

### CODE REVIEW

**Review by hernanarber — APPROVED** [review]
Looks Good.

File: app/components/modals/listingNft.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+        initialize();
+    }, [wallet, connection, provider, sdk]);
+
+    const sign = async (sellerAddress: PublicKey) => {
+        // const sol = price * LAMPORTS_PER_SOL;
+
+        console.log("creating tx");
+        if (!sdk) {
+            throw new WalletNotConnectedError();
```
Comment (jim4067) [inline app/components/modals/listingNft.tsx:49]: I think this is the wrong error to throw. Why do we have this check when we have one for `wallet`?

File: app/components/modals/listingNft.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        }
+
+        const ix = await sdk.createListing(mint, price).then((ix) => {
```
Comment (jim4067) [inline app/components/modals/listingNft.tsx:52]: I'd prefer if we used `async-await` going forward, but this works

File: app/components/modals/listingNft.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            console.log("ix 📝: ", ix);
+            console.log("to sale at: ", price);
+            return ix;
+        });
+
+        if (!ix) {
+            throw new Error();
+        }
```
Comment (jim4067) [inline app/components/modals/listingNft.tsx:60]: Do we need this? The SDK should take care of this

File: app/components/modals/listingNft.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+        const tx = new Transaction().add(ix);
+
+        if (!tx) {
+            throw new Error();
+        }
```
Comment (jim4067) [inline app/components/modals/listingNft.tsx:66]: I don't think we need this too

File: app/components/modals/listingNft.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+        if (!provider) {
+            throw new Error();
+        }
```
Comment (jim4067) [inline app/components/modals/listingNft.tsx:70]: We are already doing this check above 

```ts
if (!provider && !sdk)
```

File: app/components/modals/listingNft.tsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    if (price && nftAddress && sdk) {
+        try {
+            if (!sellerAddress) {
+                throw new Error();
+            }
+            sign(sellerAddress);
+        } catch (error) {
+            console.error("Error creating listing:");
+        }
+    }
```
Comment (jim4067) [inline app/components/modals/listingNft.tsx:119]: why are we signing here if we've already listed?

File: app/explore/data/tracks.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
+import axios from "axios";
+
+const apiUrl =
+    "https://ageless-period-400708.uc.r.appspot.com/api/v1/nfts/soundwork";
+
```
Comment (jim4067) [inline app/explore/data/tracks.tsx:5]: We shouldn't be doing this here

**Review by jim4067 — CHANGES_REQUESTED** [review]

File: app/explore/data/tracks.tsx (hunk 7 above)
Comment (kasuba97) [reply app/explore/data/tracks.tsx:5]: yes, cleaning 😊

File: app/components/modals/listingNft.tsx (hunk 6 above)
Comment (kasuba97) [reply app/components/modals/listingNft.tsx:119]: we are calling sign there as it wont execute if not called.
should we put it a useEffect?

Comment (kasuba97) [reply app/components/modals/listingNft.tsx:119]: i have moved the sign prompt code and placed it in the ``useEffect`` this should be resolved now

Comment (jim4067) [reply app/components/modals/listingNft.tsx:119]: Ahh I see. 

We need to get rid of most of these checks as they're just redundant.

**Review by jim4067 — APPROVED** [review]
This works. 

let's remove the `console.log`s and merge

**Review by jim4067 — APPROVED** [review]

**Pull request merged**